src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -185,18 +200,18 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem)
 
             token.ThrowIfCancellationRequested();
 
-            // multiple loads may occur at the same time
-            if (currentLoader == _loader)
-            {
-                UpdatePackageList(loadedItems.ToList(), refresh: false);
-            }
-
-            token.ThrowIfCancellationRequested();
-
             await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
                 _loadingStatusBar.ItemsLoaded = currentLoader.State.ItemsCount;
+                
+                // multiple loads may occur at the same time
+                if (currentLoader == _loader)
+                {
+                    UpdatePackageList(loadedItems.ToList(), refresh: false);
```

Comment:
i have made a bunch of improvements all across to reduce calls to ToList and enumerating, in general.